### PR TITLE
sailor should support dynamic flow control

### PR DESCRIPTION
**Pull request by irisked**

### CODE REVIEW

File: lib/amqp.js [hunk 1]
```diff
@@ -165,24 +186,8 @@ class Amqp {
...
-        };
-
-        try {
-            const result = await promisifiedPublish(exchangeName, routingKey, payloadBuffer, options);
-            if (!result) {
-                log.warn('Buffer full when publishing a message to '
-                    + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
-            }
-            return result;
         } catch (error) {
```
Comment (VitaliyKovalchuk) [inline lib/amqp.js:189]: what this `catch` do?

File: lib/amqp.js [hunk 2]
```diff
@@ -138,16 +138,37 @@ class Amqp {
...
+                log.warn('Buffer full when publishing a message to '
+                    + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
+            }
+            return result;
+        } catch (error) {
+            log.error(error, 'Failed on publishing message to queue');
+            const currentDelay = settings.AMQP_PUBLISH_RETRY_DELAY;
+            const maxDelay = settings.AMQP_PUBLISH_MAX_RETRY_DELAY;
+            const delay = Math.min(currentDelay * iteration, maxDelay);
+            await new Promise(resolve => setTimeout(resolve, delay));
```
Comment (VitaliyKovalchuk) [inline lib/amqp.js:158]: please, add some `trace` log message with `iteration`, `delay`, .... for easy debugging.

**Comment by anton-kotenko** [comment]
merged here https://github.com/elasticio/sailor-nodejs/pull/139

**Comment by anton-kotenko** [comment]
https://github.com/elasticio/sailor-nodejs/pull/139

**Pull request closed without merging**